File: app/src/main/java/com/boardgamegeek/pref/VersionPreference.java
package com.boardgamegeek.pref;

import android.content.Context;
import android.preference.Preference;
import android.util.AttributeSet;

import com.boardgamegeek.util.HelpUtils;

public class VersionPreference extends Preference {
	public VersionPreference(Context context, AttributeSet attrs) {
		super(context, attrs);
	}

	@Override
	public CharSequence getSummary() {
		return HelpUtils.getVersionName(getContext());
	}
}

File: app/src/main/java/com/boardgamegeek/pref/VersionPreference.kt
package com.boardgamegeek.pref

import android.content.Context
import android.util.AttributeSet
import androidx.preference.Preference
import com.boardgamegeek.extensions.versionName

class VersionPreference(context: Context, attrs: AttributeSet) : Preference(context, attrs) {
    override fun getSummary(): CharSequence {
        return context.versionName()
    }

    override fun isSelectable(): Boolean {
        return false
    }
}
